TESTS/mbed_drivers/watchdog_reset/main.cpp

```diff
@@ -176,27 +185,24 @@ void test_deepsleep_reset()
     }
 
     // Phase 1. -- run the test code.
-    Semaphore sem(0, 1);
-    LowPowerTimeout lp_timeout;
-    if (send_reset_notification(&current_case, 2 * TIMEOUT_MS) == false) {
+    if (send_reset_notification(&current_case, 2 * TIMEOUT_MS + SERIAL_FLUSH_TIME_MS) == false) {
         TEST_ASSERT_MESSAGE(0, "Dev-host communication error.");
         return;
     }
+    wait_us(SERIAL_FLUSH_TIME_US); // Wait for the serial buffers to flush.
     Watchdog &watchdog = Watchdog::get_instance();
     TEST_ASSERT_FALSE(watchdog.is_running());
     TEST_ASSERT_TRUE(watchdog.start(TIMEOUT_MS));
     TEST_ASSERT_TRUE(watchdog.is_running());
-    // Watchdog should fire before twice the timeout value.
-    lp_timeout.attach_us(mbed::callback(release_sem, &sem), 1000ULL * (2 * TIMEOUT_MS));
-    wait_ms(SERIAL_FLUSH_TIME_MS); // Wait for the serial buffers to flush.
     if (!sleep_manager_can_deep_sleep()) {
         TEST_ASSERT_MESSAGE(0, "Deepsleep should be allowed.");
     }
-    sem.wait(); // Device reset expected.
+    // Watchdog should fire before twice the timeout value.
+    ThisThread::sleep_for(2 * TIMEOUT_MS); // Device reset expected.
```

Comment:
According to the K64F spec, this wait has to be updated. A watchdog reset during a deepsleep on this target is expected to happen exactly after twice the timeout value -- #11774.